File: src/checkers-api/Models/Responses/RoomInfo.cs
using checkers_api.Models.GameModels;

namespace checkers_api.Models.Responses;

public class RoomInfo
{
    public string RoomId { get; }
    public Player RoomOwner { get; }
    public Player? RoomGuest { get; }

    public RoomInfo(string roomId, Player roomOwner, Player? roomGuest = null)
    {
        RoomId = roomId;
        RoomOwner = roomOwner;
        RoomGuest = roomGuest;
    }
}

File: src/checkers-client/src/models/room/roomInfo.ts
import Player from "../game/player";

export default class RoomInfo {
  public readonly roomId: string;
  public readonly roomOwner: Player;
  public readonly roomGuest?: Player;

  constructor (roomId: string, roomOwner: Player, roomGuest?: Player) {
    this.roomId = roomId;
    this.roomOwner = roomOwner;
    this.roomGuest = roomGuest;
  }
}
